src/status_im/browser/multi_chain.cljs

```diff
@@ -0,0 +1,160 @@
+  (= type constants/multi-chain-add-ethereum-chain))
+
+(defn get-switch-ethereum-chain-permission [params]
+  {:type :switch-ethereum-chain
+   :title "Allow this site to switch the network?"
+   :description "This will switch the selected network within Status to a previously added network:\n${{network-from}} -> ${{network-to}}\nit will require login/logout"
+   :params params
+   :icon :main-icons/wallet})
+
+(defn get-add-ethereum-chain-permission [params]
+  {:type :add-ethereum-chain
+   :title "Allow this site to add a network?"
+   :description "This will allow this network to be used within Status.Status does not verify custom networks.Learn about scams and network security risks."
+   :params params
+   :icon :main-icons/wallet})
+
+(defn- hide-sheet-and-dispatch [event]
+  (re-frame/dispatch [:bottom-sheet/hide])
+  (re-frame/dispatch event))
+
+(defn- call-permission-callback
+  [allowed? type message-id params]
+  (let [result {:allowed? allowed?
+                :message-id message-id
+                :params params}]
+    (cond
+      (switch-ethereum-chain? type)
+      (hide-sheet-and-dispatch [:multi-chain/permissioned-switch-ethereum-chain result])
+
+      (add-ethereum-chain? type)
+      (hide-sheet-and-dispatch [:multi-chain/permissioned-add-ethereum-chain result]))))
+
+(fx/defn allow-permission
+  {:events [:browser.multi-chain.ui/dapp-permission-allowed]}
+  [{:keys [db] :as cofx} message-id requested-permission params]
+  (call-permission-callback true requested-permission message-id params))
+
+(fx/defn deny-permission
+  {:events [:browser.multi-chain.ui/dapp-permission-denied]}
+  [{:keys [db] :as cofx} message-id requested-permission params]
+  (call-permission-callback false requested-permission message-id params))
+
+(fx/defn permissioned-switch-ethereum-chain
+  {:events [:multi-chain/permissioned-switch-ethereum-chain]}
+  [cofx {:keys [allowed? message-id params]}]
+  (if allowed?
+    (let [{:keys [target-network-id]} params]
+      (debounce/dispatch-and-chill [::network/connect-network-pressed target-network-id] 1000))
+    (send-permission-reject-error-to-bridge cofx message-id)))
+
+(fx/defn permissioned-add-ethereum-chain
+  {:events [:multi-chain/permissioned-add-ethereum-chain]}
+  [{:keys [db] :as cofx} {:keys [allowed? message-id params]}]
+  (if allowed?
+    (let [{:keys [new-networks id]} params]
+      {:db (assoc db :networks/networks new-networks)
+       ::json-rpc/call [{:method "settings_saveSetting"
+                         :params [:networks/networks (vals new-networks)]
+                         :on-success #(re-frame/dispatch [:multi-chain/send-success-call-to-bridge cofx id message-id])
+                         :on-error   #(log/error "failed to perform settings_saveSetting" %)}]})
+    (send-permission-reject-error-to-bridge cofx message-id)))
+
+(defn- process-switch-ethereum-permission
+  [dapp-name permission message-id params]
+  (when (wallet-connected?)
```

Comment:
you shouldn't use a subscription inside the event, you must get all data from event's db